### PR TITLE
fix: tag wasn't able to use default html-tags like `button`, `a`, `input`

### LINKED ISSUE #2678: DBTag [removeable]: includes a lot of unused/not displayed DOM from DBCheckbox
### Which generators are impacted?

- [ ] All
- [ ] HTML
- [X] React
- [ ] Angular
- [ ] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

Currently DB Tag with prop "removeable"  also includes a lot of not displayed HTML Code generated by checkboxes' validation (<DBInfotext>) – is this really necessary?

````
<div class="db-tag" data-semantic="successful" data-icon="plus">
     <div class="db-checkbox">
           <label for="checkbox">
                <input type="checkbox">
                Label-Text
           </label>
         <span id="checkbox-valid-message" class="db-infotext" data-semantic="successful" data-size="small">TODO: Add a validMessage</span>
         <span id="checkbox-invalid-message" class="db-infotext" data-semantic="critical" data-size="small">TODO: Add an invalidMessage</span>
    </div>
</div>
````

### Expected Behaviour

When using DB Tag with removeable option, a lighter version of DBCheckbox should be rendered

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #2678

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/components/tag/tag.lite.tsx [hunk 1]
```diff
@@ -41,16 +41,17 @@ export default function DBTag(props: DBTagProps) {
...
-					noText
-					variant="ghost"
+					data-icon="cross"
+					data-size="small"
+					data-no-text="true"
+					data-variant="ghost"
 					title={state.getRemoveButtonText()}>
 					{state.getRemoveButtonText()}
-				</DBButton>
+				</button>
```
Comment (mfranzke) [inline packages/components/src/components/tag/tag.lite.tsx:54]: we would need to add migration guide entries for changes like this.

Comment (nmerget) [reply packages/components/src/components/tag/tag.lite.tsx:54]: Why? This is only the internal component. For pure HTML users nothing would change, as well for framework users.

File: showcases/shared/tag.json [hunk 2]
```diff
@@ -142,41 +144,38 @@
...
-					{
-						"name": "checkbox"
-					}
-				]
+				"code": {
+					"angular": "<db-tag><label><input type=\"checkbox\"/>Interactive Checkbox</label></db-tag>",
+					"html": "<div class=\"db-tag\"><label><input type=\"checkbox\"/>Interactive Checkbox</label></div>",
+					"react": "<DBTag><label><input type=\"checkbox\"/>Interactive Checkbox</label></DBTag>",
+					"vue": "<DBTag><label><input type=\"checkbox\"/>Interactive Checkbox</label></DBTag>"
+				}
```
Comment (mfranzke) [inline showcases/shared/tag.json:152]: @nmerget to my understanding, we could use children.name only to reference DB UI components or native HTML elements, but none of those native nested tags constructs like e.g. `<label><input type=\"checkbox\"/>Interactive Checkbox</label>`, could we?

Comment (mfranzke) [reply showcases/shared/tag.json]: actually, it is, via nested `children`

File: showcases/patternhub/scripts/utils.js [hunk 3]
```diff
@@ -193,6 +193,10 @@ export const getCodeByFramework = (
 		}
 	}
 
+	if (native && name === 'input') {
```
Comment (mfranzke) [inline showcases/patternhub/scripts/utils.js:196]: why do you specifically select for `input` here?

Comment (nmerget) [reply showcases/patternhub/scripts/utils.js:196]: It has to be `<input/>` and nextjs throws and build error if we use `<input></input>`

File: packages/components/src/components/tag/tag.lite.tsx [hunk 4]
```diff
@@ -25,6 +32,26 @@ export default function DBTag(props: DBTagProps) {
...
+					'button:not(.db-tab-remove-button)'
+				);
+				const input: HTMLInputElement = ref?.querySelector('input');
+				for (const element of [button, input]) {
+					if (element) {
+						element.disabled = props.disabled;
+					}
+				}
+			}
+		}
```
Comment (mfranzke) [inline packages/components/src/components/tag/tag.lite.tsx:52]:
```suggestion
		if (state.initialized && ref && props.disabled !== undefined) {
		  const button: HTMLButtonElement = ref?.querySelector(
			  'button:not(.db-tab-remove-button)'
		  );
		  const input: HTMLInputElement = ref?.querySelector('input');
		  for (const element of [button, input]) {
			  if (element) {
				  element.disabled = props.disabled;
			  }
		  }
	    }
```

**Review by mfranzke — APPROVED** [review]

**Review by mfranzke — APPROVED** [review]

**Pull request merged**